ghidra_jupyter/src/ghidra_jupyter/installer.py

```diff
@@ -47,20 +49,34 @@ def _get_ghidra_dir(ghidra_install_dir: Optional[str]) -> str:
     return ghidra_install_dir or os.environ.get("GHIDRA_INSTALL_DIR")
 
 
-def get_download_url(repo, name_pattern):
+def _get_ghidra_version(ghidra_install_dir: Optional[str]) -> version.Version:
+    app_properties = os.path.join(ghidra_install_dir, "Ghidra", "application.properties")
+    with open(app_properties, "r") as f:
+        for line in f.readlines():
+            key, value = line.split("=")
+            if key == "application.version":
+                return version.parse(value)
+
+
+def get_download_url(repo, name_pattern) -> Tuple[str, version.Version]:
     release = requests.get(f"https://api.github.com/repos/{repo}/releases/latest")
     release.raise_for_status()
     for asset in release.json()["assets"]:
-        if re.match(name_pattern, asset["name"]):
-            return asset["browser_download_url"]
+        m = re.match(name_pattern, asset["name"])
+        if m:
+            extension_version = version.parse(m.group(1))
+            return asset["browser_download_url"], extension_version
+
```

Comment:
My automatic concern is that these may return `None`. But I guess this doesn't matter too much - worst case, the installer fails.

Reply:
Well, this PR is an improvement from "`get_download_url` always returns `None` because the regex doesn't match" to "`get_download_url` might under unknown circumstances return `None`", so I am just going to treat it as a definitive improvement